x-pack/plugins/observability_solution/observability/public/hooks/slo/use_fetch_synthetics_suggestions.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useKibana } from '../../utils/kibana_react';
+
+export interface Suggestion {
+  label: string;
+  value: string;
+  count: number;
+}
+
+export interface UseFetchSyntheticsSuggestions {
+  suggestions: Suggestion[];
+  isLoading: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+}
+
+export interface Params {
+  fieldName: string;
+  filters?: {
+    locations?: string[];
+    monitorIds?: string[];
+    tags?: string[];
+    projects?: string[];
+  };
+  search: string;
+}
+
+type ApiResponse = Record<string, Suggestion[]>;
+
+export function useFetchSyntheticsSuggestions({
+  filters,
+  fieldName,
+  search,
+}: Params): UseFetchSyntheticsSuggestions {
+  const { http } = useKibana().services;
+  const { locations, monitorIds, tags, projects } = filters || {};
+
+  const { isInitialLoading, isLoading, isError, isSuccess, isRefetching, data } = useQuery({
+    queryKey: ['fetchSyntheticsSuggestions', locations, monitorIds, tags, projects, search],
+    queryFn: async ({ signal }) => {
+      try {
+        const suggestions = await http.get<ApiResponse>('/internal/synthetics/suggestions', {
+          query: {
+            locations: locations || [],
+            monitorQueryIds: monitorIds || [],
+            tags: tags || [],
+            projects: projects || [],
+            query: search,
+          },
+          signal,
+        });
+
+        return suggestions;
+      } catch (error) {
+        // ignore error
+      }
+    },
+    refetchOnWindowFocus: false,
+    keepPreviousData: true,
+  });
+
+  return {
+    suggestions: isInitialLoading ? [] : data?.[fieldName] ?? [],
+    isLoading: isInitialLoading || isLoading || isRefetching,
```

Comment:
don't understand this, won't isLoading always true in those cases. not sure why we need this OR condition.

Reply:
react query v4 is a bit complicated with the isLoading, isInitialLoading, isRefetching. They solve this in v5.

Here since we are not using refetch() nor automatic refresh. We can remove the need for `isRefetching`, and since we don't need to handle initial vs non initial loading differently, we can just use `isLoading`. 
I think there is other hook in the slo hook folder that could be simplified a bit.

As for `suggestions`, I think we can simplify to: `suggestions: data?.[fieldName] ?? []`